core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -1068,6 +1096,104 @@ public void run() {
     }
   }
 
+  private static RetryPolicies getRetryPolicies(Attributes config) {
+    JsonObject serviceConfig = config.get(GrpcAttributes.NAME_RESOLVER_ATTR_SERVICE_CONFIG);
+    return getRetryPolicies(serviceConfig);
+  }
+
+  // TODO(zdapeng): maybe move it to a more appropriate place
```

Comment:
+1 (but optional, because I'm not sure where would be more appropriate, but it seems like it might be nice to keep the json parsing in a separate location)